### PR TITLE
Refactoring Delphes module

### PR DESCRIPTION
**Pull request by jlingema**
As discussed with @drasal, I did some changes to the Delphes module in FCCSW: Most notably, I removed the HepMC reader functionality that is already provided by the Generation package and moved the EDM conversion into dedicated tools to allow the user to select only the output that is needed. 

The results were tested against the output of the old version of the code, when using the same rnd-number seeds in Delphes and Pythia. For reference, I used [this](https://github.com/jlingema/fcc-physics/blob/delphes/example/compare_delphes_trees.py) to compare the EDM output files.

### CODE REVIEW

File: Sim/SimDelphesInterface/options/PythiaDelphes_config.py [hunk 1]
```diff
@@ -55,25 +57,58 @@
...
+jetSaveTool.DataOutputs.jetConstituents.Path = "jetParts"
+jetSaveTool.DataOutputs.jetConstituentAssociations.Path = "jetsToParts"
+jetSaveTool.DataOutputs.jetFlavorTags.Path = "jetsFlavor"
+jetSaveTool.DataOutputs.jetFlavorTagAssociations.Path = "jetsToFlavor"
+jetSaveTool.DataOutputs.bTags.Path = "bTags"
+jetSaveTool.DataOutputs.cTags.Path = "cTags"
+jetSaveTool.DataOutputs.tauTags.Path = "tauTags"
+jetSaveTool.DataOutputs.jetBTagAssociations.Path = "jetsToBTags"
+jetSaveTool.DataOutputs.jetCTagAssociations.Path = "jetsToCTags"
+jetSaveTool.DataOutputs.jetTauTagAssociations.Path = "jetsToTauTags"
```
Comment (hegner) [inline Sim/SimDelphesInterface/options/PythiaDelphes_config.py]: Any chance of rationalizing that in a clever way? Looks repetitive.

**Comment by hegner** [comment]
If you do not want to have any other work in the pipeline for that PR I am happy to merge.

File: Sim/SimDelphesInterface/src/DelphesSimulation.cpp [hunk 2]
```diff
@@ -193,24 +101,21 @@ StatusCode DelphesSimulation::initialize() {
...
-  m_chargedOutArray  = nullptr;
-  m_neutralOutArray  = nullptr;
-  m_photonOutArray   = nullptr;
-  m_jetOutArray      = nullptr;
-  m_metOutArray      = nullptr;
-  m_shtOutArray      = nullptr;
 
   m_eventCounter     = 0;
 
   if (m_outRootFile!=nullptr) m_treeWriter->Clear();
```
Comment (jlingema) [inline Sim/SimDelphesInterface/src/DelphesSimulation.cpp]: Also should apply the same bugfix as in #97 here, when resolving conflicts.

**Comment by jlingema** [comment]
retest this please

**Pull request merged**